stablehlo/dialect/TypeInference.cpp

```diff
@@ -199,9 +199,28 @@ int64_t stridedBound(int64_t bound, int64_t windowSize, int64_t stride) {
   return (bound - windowSize) / stride + 1;
 }
 
-LogicalResult verifyBatchNorm(Optional<Location> location, Value operand,
-                              Value scale, int64_t feature_index) {
-  auto operandType = operand.getType().cast<RankedTensorType>();
+LogicalResult verifyPairwiseCompatibleShapes(TypeRange values) {
+  for (auto type1 : values)
+    for (auto type2 : values)
+      if (failed(verifyCompatibleShape(type1, type2))) return failure();
+  return success();
+}
+
+LogicalResult verifyBatchNorm(Optional<Location> location,
+                              ValueRange multiDimOperands,
+                              ValueRange singleDimOperands,
+                              int64_t feature_index) {
+  if (failed(verifyPairwiseCompatibleShapes(multiDimOperands.getTypes())))
+    return emitOptionalError(
+        location,
+        "expects multi-dimensional operands to have compatible shapes.");
+
+  if (failed(verifyPairwiseCompatibleShapes(singleDimOperands.getTypes())))
+    return emitOptionalError(
+        location,
+        "expects single-dimensional operands to have compatible shapes.");
+
+  auto operandType = multiDimOperands[0].getType().cast<RankedTensorType>();
```

Comment:
Same suggestion regarding renaming as in the functions below. Given that we have the new nomenclature that involves single-dim and multi-dim operands, let's change all the old names to follow it. Otherwise, stuff like `operandType` and `scaleShape` looks out of place.